src/fides/api/api/v1/endpoints/messaging_endpoints.py

```diff
@@ -565,4 +588,190 @@ def update_basic_messaging_templates(
                 )
             )
 
-    return BulkPutMessagingTemplateResponse(succeeded=succeeded, failed=failed)
+    return BulkPutBasicMessagingTemplateResponse(succeeded=succeeded, failed=failed)
+
+
+@router.get(
+    MESSAGING_TEMPLATES_SUMMARY,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=Page[MessagingTemplateWithPropertiesSummary],
+)
+def get_property_specific_messaging_templates_summary(
+    *, db: Session = Depends(deps.get_db), params: Params = Depends()
+) -> AbstractPage[MessagingTemplate]:
+    """
+    Returns all messaging templates, automatically saving any missing message template types to the db.
+    """
+    # First save any missing template types to db
+    save_defaults_for_all_messaging_template_types(db)
+    ordered_templates = MessagingTemplate.query(db=db).order_by(
+        MessagingTemplate.created_at.desc()
+    )
+    # Now return all templates
+    return paginate(
+        ordered_templates,
+        params=params,
+    )
+
+
+@router.get(
+    MESSAGING_TEMPLATE_DEFAULT_BY_TEMPLATE_TYPE,
+    dependencies=[Security(verify_oauth_client, scopes=[MESSAGING_TEMPLATE_UPDATE])],
+    response_model=MessagingTemplateDefault,
+)
+def get_default_messaging_template(
+    template_type: MessagingActionType,
```

Comment:
Using `MessagingActionType` for this input is too open since it's possible for the user to select a message type that doesn't have a default value and this would return a 500 error. We should either:
- Return a message saying that specific message types don't have a default value
- Use a more restrictive type and update the docstring to reflect that this endpoint will only return defaults for the configurable messaging templates

The constant you defined [here](https://github.com/ethyca/fides/pull/4986/files#diff-591f5709c6b30f40e4b4f1e97bad0be6e40bb3f4e89bca18d5715c57e7278480R69) would be perfect for this use case if you converted it to an enum

Reply:
Ah yes, our CRUD layer raises a `MessagingTemplateValidationException` and I've forgotten to handle this at the endpoint layer, good catch!

Reply:
I think a 400 makes sense since it's technically an invalid / bad request